src/printer.js

```diff
@@ -1541,10 +1548,10 @@ function genericPrintNoParens(path, options, print) {
       return printFlowDeclaration(path, [
         "module.exports",
         path.call(print, "typeAnnotation"),
-        ";"
+        semi
       ]);
     case "DeclareVariable":
-      return printFlowDeclaration(path, ["var ", path.call(print, "id"), ";"]);
+      return printFlowDeclaration(path, ["var ", path.call(print, "id"), semi]);
```

Comment:
@mroch, would you be able to shed some light here? This pull request attempts to implement no-semi on prettier and we're not 100% sure how flow type declarations interact with automatic-semicolon-insertion.